docs/StardustDocs/snippets/org.jetbrains.kotlinx.dataframe.api.ColumnsSelectionDsl.DslGrammar.ColumnGroupPartOfGrammar.ForHtml.html

```diff
@@ -0,0 +1,48 @@
</p><p>    <code>| </code><strong><code>.</code></strong><code><strong><code>drop</code></strong></code><code>(</code><code><strong><code>Last</code></strong></code><code>)</code><code><strong><code>ColsWhile</code></strong></code><strong><code> { </code></strong><code><code>condition</code></code><strong><code> }</code></strong></p><p>    <code>| </code><code><strong><code>exceptNew</code></strong></code><strong><code> { </code></strong><code><code>colsSelector</code></code><strong><code> } EXPERIMENTAL!</code></strong></p><p>    <code>| </code><code><strong><code>exceptNew</code></strong></code><strong><code>(</code></strong><code><code>columnNoAccessor</code></code><strong><code>,</code></strong><code> ..</code><strong><code>) EXPERIMENTAL!</code></strong></p><p>    <code>| (</code><strong><code>.</code></strong><code><strong><code>firstCol</code></strong></code><code>| </code><strong><code>.</code></strong><code><strong><code>lastCol</code></strong></code><code>| </code><strong><code>.</code></strong><code><strong><code>singleCol</code></strong></code><code>) [ </code><strong><code>{ </code></strong><code><code>condition</code></code><strong><code> }</code></strong><code> ]</code></p><p>    <code>| </code><strong><code>.</code></strong><code><strong><code>select</code></strong></code><strong><code> { </code></strong><code><code>colsSelector</code></code><strong><code> }</code></strong></p><p>    <code>| </code><strong><code>.</code></strong><code><strong><code>take</code></strong></code><code>(</code><code><strong><code>Last</code></strong></code><code>)</code><code><strong><code>Cols</code></strong></code><strong><code>(</code></strong><code><code>number</code></code><strong><code>)</code></strong></p><p>    <code>| </code><strong><code>.</code></strong><code><strong><code>take</code></strong></code><code>(</code><code><strong><code>Last</code></strong></code><code>)</code><code><strong><code>ColsWhile</code></strong></code><strong><code> { </code></strong><code><code>condition</code></code><strong><code> }</code></strong></p><p>    </p><p><code><code>singleColumn</code></code></p><p>    <strong><code>.</code></strong><code><strong><code>colsOf</code></strong></code><strong><code>&lt;</code></strong><code><code>T</code></code><strong><code>&gt;</code></strong><code> [ </code><strong><code>(</code></strong><code><code>kType</code></code><strong><code>)</code></strong><code> ] [ </code><strong><code>{ </code></strong><code><code>condition</code></code><strong><code> }</code></strong><code> ]</code></p><p>    </p><p><code><code>columnGroupReference</code></code></p><p>    <strong><code>.</code></strong><code><strong><code>colsOf</code></strong></code><strong><code>&lt;</code></strong><code><code>T</code></code><strong><code>&gt;(</code></strong><code><code>kType</code></code><strong><code>)</code></strong><code> [ </code><strong><code>{ </code></strong><code><code>condition</code></code><strong><code> }</code></strong><code> ]</code></p></body>
```

Comment:
How it was generated? If I need somehow to change it in the future, how I could do this? Is it a result of preprocessor work?

Reply:
Yep, all we need is the `@ExportAsHtml` annotation on a KDoc-holding element and the preprocessor will generate a file like this in the `snippets` folder :).

If you change something in the KDoc of that interface and run the preprocessor like normal, the html file will be updated as well automatically.

Like:
https://github.com/Kotlin/dataframe/blob/621f1e45ff8b80a66e75b4ad59903db72ad83861/core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/api/ColumnsSelectionDsl.kt#L307